src/main/scala/com/preprocessing/edgeList/Compressor.scala

```diff
@@ -0,0 +1,177 @@
+package com.preprocessing.edgeList
+
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkConf
+import org.apache.spark.rdd.RDD
+
+object Compressor {
+  """
+Usage:
+runMain com.preprocessing.edgeList.Compressor
+--nNodes:         number of vertices in the graph
+--nEdges:         number of edges in the graph
+--inputFilename:  path to input (uncompressed, unordered) graph
+--outputFilename: output path
+--sep:            separator used in input edge list (e.g. " ", ",",  "\t")
+
+runMain com.preprocessing.edgeList.Compressor --nNodes 5 --nEdges 7 --inputFilename "src/main/resources/gaps" --outputFilename "src/main/resources/reset" --sep " "
+
+runMain com.preprocessing.edgeList.Compressor --nNodes 1005 --nEdges 25571 --inputFilename "src/main/resources/graphs/email-Eu-core/orig.net" --outputFilename "src/main/resources/graphs/email-Eu-core/reset" --sep " "
+
+Sort an input edge list by ascending source id. For each source id, the destination ids are also sorted in
+ascending order.
+Self directed edges are removed.
+Multiple edges between the same pair of nodes are removed.
+
+e.g.
+Given an unordered edge list with "gaps":
+  0 1
+  0 0
+  0 3
+  0 4
+  0 2
+  3 1
+  4 0
+  3 3
+  3 0
+
+The edgeList.Compressor will produce
+  0 1
+  0 2
+  0 3
+  0 4
+  3 0
+  3 1
+  4 0
+
+"""
+  val usage =
+    """
+      |[--nNodes int] [--nEdges int] [--inputFilename str] [--outputFilename str]
+      |""".stripMargin
+
+  def main(args: Array[String]): Unit = {
+
+    val appName: String = "edgeList.Compressor.Driver"
+
+    val conf = new SparkConf()
+      .setAppName(appName)
+      .setMaster("local[*]")
+    val sc = new SparkContext(conf)
+
+    var nNodes = 0
+    var nEdges = 0
+    var infile = ""
+    var outfile = ""
+    var sep = ""
+
+    args.sliding(2, 2).toList.collect {
+      case Array("--nNodes", argNNodes: String) => nNodes = argNNodes.toInt
+      case Array("--nEdges", argNEdges: String) => nEdges = argNEdges.toInt
+      case Array("--inputFilename", argInFile: String) => infile = argInFile
+      case Array("--outputFilename", argOutFile: String) => outfile = argOutFile
+      case Array("--sep", argSep: String) => sep = argSep
+    }
+
+    var argStr = ""
+    argStr += s"$nNodes\t"
+    argStr += s"$nEdges\t"
+    argStr += s"$infile\t"
+    argStr += s"$outfile\t"
+    println(s"args: $argStr")
+
+    var resetIndex = 0
+
+    // a map between the original vertex id to the vertex id in the compressed, sorted representation
+    // TODO; map may contain large amount of vertices; can/should it be distributed?
+    // var permutationPairRDD = RDD[(Int, Int)]()?
+    val permutationMap = collection.mutable.Map[Int, Int]()
+
+    def readEdgeList(): RDD[(Int, Int)] = {
+      val distFile: RDD[String] = sc.textFile(infile)
+
+      val edgeList: RDD[(Int, Int)] = distFile
```

Comment:
For future reference, we'll handle weighted graphs from here onward to have 3 ints instead of 2.

Reply:
I'll add a todo here;